From a48d588e9e5094f74aca868a41000d6c22ae1333 Mon Sep 17 00:00:00 2001
From: Stephen Barber <smbarber@chromium.org>
Date: Mon, 28 Aug 2017 16:55:42 -0700
Subject: [PATCH] CHROMIUM: usb: quirks: add NO_LPM quirk for plankton

LPM on plankton appears to cause xHCI host controllers to claim
that there isn't enough bandwidth to support additional devices.

BUG=b:65034294,chromium:758293
TEST=while using plankton capture, plug in USB speaker and make sure
there are no bandwidth errors in dmesg

Change-Id: Ibfcf68720b17c6b6cda107692fcf1bba36516121
Signed-off-by: Stephen Barber <smbarber@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/639327
Reviewed-by: Julius Werner <jwerner@chromium.org>
---
 drivers/usb/core/quirks.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/core/quirks.c b/drivers/usb/core/quirks.c
index 3116edfcdc18..20a068e82213 100644
--- a/drivers/usb/core/quirks.c
+++ b/drivers/usb/core/quirks.c
@@ -206,6 +206,9 @@ static const struct usb_device_id usb_quirk_list[] = {
 	{ USB_DEVICE(0x1532, 0x0116), .driver_info =
 			USB_QUIRK_LINEAR_UFRAME_INTR_BINTERVAL },
 
+	/* Google - Plankton */
+	{ USB_DEVICE(0x18d1, 0x501e), .driver_info = USB_QUIRK_NO_LPM },
+
 	/* BUILDWIN Photo Frame */
 	{ USB_DEVICE(0x1908, 0x1315), .driver_info =
 			USB_QUIRK_HONOR_BNUMINTERFACES },
-- 
2.20.1

